src/pages/workspaces/workspace/tools/WorkflowView.js

```diff
@@ -414,56 +423,54 @@ const WorkflowView = _.flow(
     try {
       const ws = Workspaces.workspace(namespace, name)
 
-      const [entityMetadata, validationResponse] = await Promise.all([
+      const [entityMetadata, validationResponse, config] = await Promise.all([
         ws.entityMetadata(),
-        ws.methodConfig(workflowNamespace, workflowName).validate()
+        this.getValidation(),
+        ws.methodConfig(workflowNamespace, workflowName).get()
       ])
-      const { methodConfiguration: config } = validationResponse
+      const isRedacted = !validationResponse
       const methods = await Methods.list({ namespace: workflowNamespace, name: workflowName })
       const snapshotIds = _.map(m => _.pick('snapshotId', m).snapshotId, methods)
-      const inputsOutputs = await Methods.configInputsOutputs(config)
+      const inputsOutputs = !isRedacted && await Methods.configInputsOutputs(config)
```

Comment:
TOL: Instead of short circuiting here, what if we generated the fake I/O data here? That way much of the code below and in the render path could just proceed without modification